dbms/src/Databases/DatabaseOnDisk.cpp

```diff
@@ -141,6 +251,74 @@ void DatabaseOnDisk::createTable(
     }
 }
 
+
+void DatabaseOnDisk::createDictionary(
+    IDatabase & database,
+    const Context & context,
+    const String & dictionary_name,
+    const ASTPtr & query)
+{
+    const auto & settings = context.getSettingsRef();
+
+    /// Create a file with metadata if necessary - if the query is not ATTACH.
+    /// Write the query of `ATTACH table` to it.
+
+    /** The code is based on the assumption that all threads share the same order of operations
+      * - creating the .sql.tmp file;
+      * - adding a table to `tables`;
+      * - rename .sql.tmp to .sql.
+      */
```

Comment:
Because we have already used DDL guard lock in `InterpreterCreateQuery`. All `CREATE` queries for a single object (table or dictionary) can be executed in one thread only. We don't need additional locking here. Maybe this comment is confusing.